clients/src/main/java/org/apache/kafka/common/record/MemoryRecordsBuilder.java

```diff
@@ -792,8 +868,13 @@ public boolean hasRoomFor(long timestamp, ByteBuffer key, ByteBuffer value, Head
 
         // For passthrough V2, ensure one producerBatch only has one DefaultRecordBatch, and since
         // in this case, DefaultRecordBatch is the value part of a DefaultRecord, so we only allow one record
-        if (magic >= RecordBatch.MAGIC_VALUE_V2 && usePassthrough)
+        // For passthrough V1, ideally we can append multiple passthrough records to the same batch, it
+        // will not work when we are migrating from V1 to V2 message format as we cannot append V1 and V2 messages
+        // in the same batch.
+        // If disableBatchingInPassthrough is true, turn off batching in passthrough mode.
+        if (usePassthrough && disableBatchingInPassthrough) {
```

Comment:
From comment it seems V2->V2 will not work with batching? If so, does this mean after both src and dst are V2, we need to change the default value of this config to true?

Reply:
@kehuum IIUC, we haven't tested passthrough with batching enabled for V2->V2, although the existing comment does sound like it may not work.
We plan to test that in BMM test pipeline. It is also possible to verify this on a locally deployed KMM pipeline (which is extra work to set up the enviornment).

Reply:
Yeah, testing v2->V2 will be step 4 of the testing plan. It should work as long as we have make sure all V1 messages are processed and no V1 messages left at source to be transmitted.